src/rendering/renderers/gpu/texture/GpuTextureSystem.ts

```diff
@@ -204,23 +205,33 @@ export class GpuTextureSystem implements System, CanvasGenerator
         return this._gpuSources[source.uid] || this.initSource(source);
     }
 
+    /**
+     * this returns s bind group for a specific texture, the bind group contains
+     * - the texture source
+     * - the texture style
+     * - the texture matrix
+     * This is cached so the bind group should only be created once per texture
+     * @param texture - the texture you want the bindgroup for
+     * @returns the bind group for the texture
+     */
     public getTextureBindGroup(texture: Texture)
     {
         return this._bindGroupHash[texture.uid] ?? this._createTextureBindGroup(texture);
     }
 
-    private _createTextureBindGroup(texture: BindableTexture)
+    private _createTextureBindGroup(texture: Texture)
     {
         const source = texture.source;
 
-        const bindGroupId = source.uid;
-
-        this._bindGroupHash[bindGroupId] = new BindGroup({
+        this._bindGroupHash[texture.uid] = new BindGroup({
             0: source,
             1: source.style,
+            2: new UniformGroup({
+                uTextureMatrix: { type: 'mat3x3<f32>', value: texture.textureMatrix.mapCoord },
```

Comment:
```suggestion
                uTextureMatrix: { type: 'mat3x3<f32>', value: texture.textureMatrix.mapCoord },
                uClampFrame: { type: 'vec4<f32>', value: texture.textureMatrix.uClampFrame },
                uClampOffset: { type: 'vec2<f32>', value: texture.textureMatrix.uClampOffset },
```

I think we should be adding the other texture uniforms here too. We'd need it for #10728 at least.

Reply:
This would break the mesh WebGPU currently as is the shader needs to have the exact uniform structure in the shader. I do see you point though. 

How about we get this one in as it fixes the rope bug and we follow with this when reviewing the Shared Shader sharing PR?